utils/config.go

```diff
@@ -274,6 +284,10 @@ func newViper(allowEnvironmentOverrides bool) *viper.Viper {
 	defaults := getDefaultsFromStruct(model.Config{})
 
 	for key, value := range defaults {
+		if key == "PluginSettings.Plugins" || key == "PluginSettings.PluginStates" {
```

Comment:
This is effectively the same change as the other review, but simplified as per the suggestion.

It's no longer necessary to munge with the environment variable parsing as we skip it entirely with the restoration of the change above.